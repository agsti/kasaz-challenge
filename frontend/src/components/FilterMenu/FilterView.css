.filter-view-container {
	position: absolute;
	top:0;
	z-index: 1;
    width: 100vw;
    display: flex;
    flex-direction: column;
    align-items: center;
	background-color: white;
	box-shadow: 0 10px 6px -6px #777;


	transition: 0.5s linear;  
	transform:translateY(0);

}

.filter-view-container.hidden {  
	transform:translateY(-200%);
 
	
}

.filter-row {
    margin: 10px 0px;
    width: 100vw;
}

.filter-min-max {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
	width: 100%;
    padding: 0 10%;
}

.filter-dropdown {
    width: 170px;
}

.filter-title {
    font-size: 1.1rem;
    text-align: center;
    margin-bottom: 10px;
}

.nrooms-list > li {
	display: inline-block;
	background-color: white;	
	padding: 10px;
	border-top: #3335 solid 1px;
	border-bottom: #3335 solid 1px;
	border-right: #3335 solid 1px;
}

.nrooms-list > li:first-child {
	border-radius: 5px 0px 0px 5px;
	border-left: #3335 solid 1px;
}

.nrooms-list > li:last-child {
	border-radius: 0px 5px 5px 0px;
}

.nrooms-list > li.selected {
	color: white;
	background-color: #41af68;
}
.see-listings {
	background-color: #68388d;
	color: white;
	width: 90%;
	text-align: center;

	padding: 10px;
	margin-left: 5px;
	margin-right: 5px;
	margin-top: 100px;
	margin-bottom: 20px;
	cursor: pointer;
}

.see-listings.disabled {
	background-color: #EEE;
	color: #777;
}
