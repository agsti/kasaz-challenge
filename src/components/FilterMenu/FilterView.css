.filter-view-container {
    width: 100vw;
    display: flex;
    flex-direction: column;
    align-items: center;

}

.filter-row {
    margin: 10px 0px;
    width: 100vw;
}

.filter-min-max {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
	width: 100%;
    padding: 0 10%;
}

.filter-min-max > select {
    width: 120px;
}

.filter-title {
    font-size: 1.1rem;
    text-align: center;
    margin-bottom: 10px;
}

.nrooms-list > li {
	display: inline-block;
	background-color: white;	
	padding: 10px;
	border-top: #3335 solid 1px;
	border-bottom: #3335 solid 1px;
	border-right: #3335 solid 1px;
}

.nrooms-list > li:first-child {
	border-radius: 5px 0px 0px 5px;
	border-left: #3335 solid 1px;
}

.nrooms-list > li:last-child {
	border-radius: 0px 5px 5px 0px;
}

.nrooms-list > li.selected {
	color: white;
	background-color: #41af68;
}
.see-listings {
	background-color: #68388d;
	color: white;
	width: calc(100% - 10px);
	text-align: center;

	padding: 10px;
	margin-left: 5px;
	margin-right: 5px;
	margin-top: 100px;
	margin-bottom: 20px;
}

