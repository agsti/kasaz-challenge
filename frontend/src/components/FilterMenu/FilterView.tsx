import React from "react";
import Select from "react-select";
import "./FilterView.css";

export type OptionsValue = {value: number, label: string};
export type Callback = (value: number) => void;
type FilterViewProps = {
	visible: boolean,
    prices: OptionsValue[], 
    sizes: OptionsValue[], 
	nRooms: OptionsValue[],

	onClickSeeListings: ()=>void,
	enableSeeListings: boolean,
	callbacks: {
		minPrice: Callback, 
		maxPrice: Callback,
		minSizeSqm: Callback,
		maxSizeSqm: Callback,
		minRooms: Callback,
	}
	
	filterValues: {
		minPrice?: OptionsValue, 
		maxPrice?: OptionsValue,
		minSizeSqm?: OptionsValue,
		maxSizeSqm?: OptionsValue,
		minRooms: OptionsValue,
	}
}

export default function FilterView(props: FilterViewProps) {
    const {
		visible,
		prices,
		sizes,
		nRooms,
		callbacks,
		filterValues,
		onClickSeeListings,
		enableSeeListings
	} = props;

	const {
		minPrice,
		maxPrice,
		minSizeSqm,
		maxSizeSqm,
		minRooms
	} = filterValues;
	
	const onSeeListingsClickHandler = ()=> {
		enableSeeListings && onClickSeeListings()
	}

    return (
        <div className={visible?"filter-view-container" : "filter-view-container hidden"}>
            <div className="filter-row">
                <div className="filter-title">
                    PRECIO
            </div>
                <div className="filter-min-max">
              <Select
					  className="filter-dropdown"
					  placeholder="Precio minimo"
					value={minPrice}
					onChange={(v:OptionsValue) => callbacks.minPrice(v.value || null)}
					options={prices}
				  />
		
              <Select
					  className="filter-dropdown"
					  placeholder="Precio maximo"
					value={maxPrice}
					onChange={(v:OptionsValue) => callbacks.maxPrice(v.value || null)}
					options={prices}
				  />
                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    TAMAÑO
            </div>

                <div className="filter-min-max">
              <Select
					  className="filter-dropdown"
					  placeholder="Tamaño minimo"
					value={minSizeSqm}
					onChange={(v:OptionsValue) => callbacks.minSizeSqm(v.value || null)}
					options={sizes}
				  />
					
              <Select
					  className="filter-dropdown"
					  placeholder="Tamaño maximo"
					value={maxSizeSqm}
					onChange={(v:OptionsValue) => callbacks.maxSizeSqm(v.value || null)}
					options={sizes}
				  />

                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    HABITACIONES
                </div>
            </div>
                <ol className="nrooms-list">
                    {
                        nRooms.map((r, i) => (
                            <li 
								className={r.value===minRooms.value ? "selected":""}
								key={r.value}
								onClick={() => {
									callbacks.minRooms(nRooms[i].value)
								}}
							>
							{r.label}
							</li>
                        ))
                    }
                </ol>
			<div className={enableSeeListings?"see-listings":"see-listings disabled"} onClick={onSeeListingsClickHandler}>
				VER INMUEBLES
			</div>
        </div>
    )

}
