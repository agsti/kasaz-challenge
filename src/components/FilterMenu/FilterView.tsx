import React from "react";
import Select from "react-select";
import "./FilterView.css";

export type OptionsValue = {value: number, label: string};
export type Callback = (value: number) => void;
type FilterViewProps = {
    prices: OptionsValue[], 
    sizes: OptionsValue[], 
    nRooms: OptionsValue[],
	callbacks: {
		minPrice: Callback, 
		maxPrice: Callback,
		minSizeSqm: Callback,
		maxSizeSqm: Callback,
		minRooms: Callback,
	}
	
	filterValues: {
		minPrice: OptionsValue, 
		maxPrice: OptionsValue,
		minSizeSqm: OptionsValue,
		maxSizeSqm: OptionsValue,
		minRooms: OptionsValue,
	}
}


export default function FilterView(props: FilterViewProps) {
    const {prices, sizes, nRooms, callbacks, filterValues} = props;
	const {
		minPrice,
		maxPrice,
		minSizeSqm,
		maxSizeSqm,
		minRooms
    } = filterValues;
    return (
        <div className="filter-view-container">
            <div className="filter-row">
                <div className="filter-title">
                    PRECIO
            </div>
                <div className="filter-min-max">
              <Select
			  		className="filter-dropdown"
					value={minPrice}
					onChange={(v:OptionsValue) => callbacks.minPrice(v.value)}
					options={prices}
				  />
			<div>
			-
			</div> 
              <Select
			  		className="filter-dropdown"
					value={maxPrice}
					onChange={(v:OptionsValue) => callbacks.maxPrice(v.value)}
					options={prices}
				  />
                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    TAMANO
            </div>

                <div className="filter-min-max">
              <Select
			  		className="filter-dropdown"
					value={minSizeSqm}
					onChange={(v:OptionsValue) => callbacks.minSizeSqm(v.value)}
					options={sizes}
				  />
					<div>
						-
					</div>
              <Select
			  		className="filter-dropdown"
					value={maxSizeSqm}
					onChange={(v:OptionsValue) => callbacks.maxSizeSqm(v.value)}
					options={prices}
				  />

                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    HABITACIONES
                </div>
            </div>
                <ol className="nrooms-list">
                    {
                        nRooms.map((r, i) => (
                            <li 
								className={r.value===minRooms.value ? "selected":""}
								key={r.value}
								onClick={() => {
									callbacks.minRooms(nRooms[i].value)
								}}
							>
							{r.label}
							</li>
                        ))
                    }
                </ol>
			<div className="see-listings">
				VER INMUEBLES
			</div>
        </div>
    )

}
