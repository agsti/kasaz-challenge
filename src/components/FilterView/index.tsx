import React from "react";

import "./FilterView.css";

type FilterViewProps = {
    prices: string[], 
    propSizes: string[], 
    nRooms: string[]
}


export default function FilterView(props: FilterViewProps) {
    const {prices, propSizes, nRooms} = props;
    return (
        <div className="filter-view-container">
            <div className="filter-row">
                <div className="filter-title">
                    PRECIO
            </div>
                <div className="filter-min-max">
                    <select id="filter-price-min">
                       <option value="" disabled selected>Precio minimo</option>
                        {
                            prices.map(p => (
                                <option value={p} key={p} >{p}</option>
                            ))
                        }
                    </select>
					<div>
						-
					</div>

                    <select id="filter-price-max">
                       <option value="" disabled selected>Precio maximo</option>
                        {
                            prices.map(p => (
                                <option value={p} key={p} >{p}</option>
                            ))
                        }
                    </select>
                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    TAMANO
            </div>

                <div className="filter-min-max">
                    <select id="filter-prop-size-min">
                       <option value="" disabled selected>Tamano minimo</option>
                        {
                            propSizes.map(p => (
                                <option value={p} key={p} >{p}</option>
                            ))
                        }
                    </select>
					<div>
						-
					</div>
                    <select id="filter-prop-size-max">
                       <option value="" disabled selected>Tamano maximo</option>
                        {
                            propSizes.map(p => (
                                <option value={p} key={p} >{p}</option>
                            ))
                        }
                    </select>

                </div>
            </div>

            <div className="filter-row">
                <div className="filter-title">
                    HABITACIONES
                </div>
            </div>
                <ol className="nrooms-list">
					<li className="selected">TODOS</li>
                    {
                        nRooms.map(r => (
                            <li key={r}>{r}</li>
                        ))
                    }
                </ol>
			<div className="see-listings">
				VER INMUEBLES
			</div>
        </div>
    )

}
