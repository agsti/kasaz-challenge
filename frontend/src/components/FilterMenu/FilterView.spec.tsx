import React from 'react';
import { shallow} from 'enzyme';
import {describe, expect, it} from '@jest/globals';

import FilterView from './FilterView';

describe('FilterView', () => {
  it('should render correctly', () => {
       const component = shallow(<FilterView 
        visible={true}
        
        prices={[
            {value: 1, label:"price 1"}
        ]}
        sizes={[
            {value: 1, label:"size 1"}
        ]}
        nRooms={[
            {value: 1, label:"room 1"}
        ]}

        onClickSeeListings= {() => {}}
        enableSeeListings={true}

        callbacks={{
            minPrice: null,
            maxPrice: null,
            minSizeSqm: null,
            maxSizeSqm: null,
            minRooms:null,
        }}

        filterValues={{
            minPrice : null, 
            maxPrice : null,
            minSizeSqm : null,
            maxSizeSqm : null,
            minRooms: {value: 1, label:"room 1"},
        }}
        validation={{
            sizesErr : false, 
            pricesErr : false
        }}
       />);
  
    expect(component).toMatchSnapshot();
  });
});
