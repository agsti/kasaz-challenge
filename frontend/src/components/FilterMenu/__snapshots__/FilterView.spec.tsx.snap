// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterView should render correctly 1`] = `
<div
  className="filter-view-container"
>
  <div
    className="filter-row"
  >
    <div
      className="filter-title"
    >
      PRECIO
    </div>
    <div
      className="filter-min-max"
    >
      <StateManager
        className="filter-dropdown"
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "price 1",
              "value": 1,
            },
          ]
        }
        placeholder="Precio minimo"
        value={null}
      />
      <StateManager
        className="filter-dropdown"
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "price 1",
              "value": 1,
            },
          ]
        }
        placeholder="Precio maximo"
        value={null}
      />
    </div>
  </div>
  <div
    className="filter-row"
  >
    <div
      className="filter-title"
    >
      TAMAÑO
    </div>
    <div
      className="filter-min-max"
    >
      <StateManager
        className="filter-dropdown"
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "size 1",
              "value": 1,
            },
          ]
        }
        placeholder="Tamaño minimo"
        value={null}
      />
      <StateManager
        className="filter-dropdown"
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "size 1",
              "value": 1,
            },
          ]
        }
        placeholder="Tamaño maximo"
        value={null}
      />
    </div>
  </div>
  <div
    className="filter-row"
  >
    <div
      className="filter-title"
    >
      HABITACIONES
    </div>
  </div>
  <ol
    className="nrooms-list"
  >
    <li
      className="selected"
      key="1"
      onClick={[Function]}
    >
      room 1
    </li>
  </ol>
  <div
    className="see-listings"
    onClick={[Function]}
  >
    VER INMUEBLES
  </div>
</div>
`;
